### PR TITLE
feat(evaluationResults): fetch the proper provider log when seeing evaluation results

### PR DESCRIPTION
**Pull request by geclos**
In evaluation results you now see the provider log that was actually evaluated rather than the last provider log of the full conversation.

### CODE REVIEW

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx [hunk 1]
```diff
@@ -18,36 +18,52 @@ import { DocumentLogInfo } from '../../../../../logs/_components/DocumentLogs/Do
...
   })
-  const { data: providerLogs } = useProviderLogs({
+  const { data: _providerLogs } = useProviderLogs({
     documentLogUuid: documentLog?.uuid,
   })
+
+  const idx = _providerLogs?.findIndex((p) => p.id === providerLogId)
+  const providerLogs = _providerLogs?.slice(0, idx + 1)
+
+  console.log(providerLogId, _providerLogs, providerLogs)
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx:65]: 🔥

Comment (csansoon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx:65]: 🔥

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx [hunk 2]
```diff
@@ -113,21 +129,24 @@ export function EvaluationResultInfo({
             {selectedTab === 'messages' && (
               <EvaluationResultMessages providerLog={providerLog} />
             )}
-            <div className='w-full flex justify-center'>
-              <Button variant='link' onClick={onClickOpen}>
-                Check original log
-                <Icon name='arrowRight' widthClass='w-4' heightClass='h-4' />
-              </Button>
-            </div>
+            {evaluationResult.evaluatedProviderLogId && (
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx:132]: So if result have `evaluatedProviderLogId` it means it has a document log evaluated? 
Why not check the `documentLog` presence?

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx (hunk 1 above)
Comment (geclos) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx:65]: thanks

Comment (geclos) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx:65]: thanks

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx (hunk 2 above)
Comment (geclos) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx:132]: we always evaluate provider logs, not document logs, so it's more accurate to only show the option to see the original log if there is an evaluated provider log

**Review by csansoon — APPROVED** [review]

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx (hunk 2 above)
Comment (andresgutgon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx:130]: I know it is an edge case, but how do evaluations work with manually uploaded logs by API? We don't have a provider log in that case do we? Only a document log? Or maybe I'm mixing things.

**Review by andresgutgon — APPROVED** [review]

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx (hunk 2 above)
Comment (geclos) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx:130]: we do have provider log for those also

**Pull request merged**